semantic/parse_test.go

```diff
@@ -98,37 +124,58 @@ func TestMustParse(t *testing.T) {
 	}
 }
 
-func TestMustParse_Panic(t *testing.T) {
-	defer func() { _ = recover() }()
-
-	semantic.MustParse("", "<unknown>")
-
-	// if we reached here, then we can't have panicked
-	t.Errorf("function did not panic when given an unknown ecosystem")
-}
-
-func TestMustParse_Debian_InvalidVersion(t *testing.T) {
-	defer func() {
-		if r := recover(); r == nil {
-			t.Errorf("expected panic, got nil")
-		}
-	}()
-
-	semantic.MustParse("1.2.3-not-a-debian:version!@#$", "Debian")
-
-	// if we reached here, then we can't have panicked
-	t.Errorf("function did not panic when given an invalid version")
-}
-
-func TestMustParse_Hackage_InvalidVersion(t *testing.T) {
-	defer func() {
-		if r := recover(); r == nil {
-			t.Errorf("expected panic, got nil")
-		}
-	}()
-
-	semantic.MustParse("1.2.3.4.5-notallowed", "Hackage")
-
-	// if we reached here, then we can't have panicked
-	t.Errorf("function did not panic when given an invalid version")
+func TestMustParse_InvalidVersions(t *testing.T) {
```

Comment:
Probably beyond the scope of this review, but is the MustParse tests necessary? I thought there's just one default implementation that's pretty simple, if err panic.

I think just testing Parse for error would be enough here.